lib/ansible/modules/cloud/vmware/vmware_dvs_portgroup.py

```diff
@@ -157,242 +217,1038 @@
-                'present': self.state_exit_unchanged,
-                'absent': self.state_create_dvspg,
+                'present': self.update_port_group,
+                'absent': self.create_port_group,
             }
         }
         try:
-            dvspg_states[self.module.params['state']][self.check_dvspg_state()]()
+            dvspg_states[self.module.params['state']][self.check_port_group_state()]()
         except vmodl.RuntimeFault as runtime_fault:
             self.module.fail_json(msg=runtime_fault.msg)
         except vmodl.MethodFault as method_fault:
             self.module.fail_json(msg=method_fault.msg)
         except Exception as e:
             self.module.fail_json(msg=str(e))
 
+    def check_port_group_state(self):
+        """Check if DVS portgroup exists"""
+        self.dv_switch = find_dvs_by_name(self.content, self.switch_name)
+
+        if self.dv_switch is None:
+            self.module.fail_json(
+                msg="A distributed virtual switch with name %s does not exist" % self.switch_name
+            )
+        self.dvs_portgroup = find_dvspg_by_name(self.dv_switch, self.portgroup_name)
+        if self.dvs_portgroup:
+            return 'present'
+        return 'absent'
+
     def create_port_group(self):
-        config = vim.dvs.DistributedVirtualPortgroup.ConfigSpec()
+        """Create DVS portgroup"""
+        changed = True
+        results = dict(changed=changed)
 
-        # Basic config
-        config.name = self.module.params['portgroup_name']
-        config.numPorts = self.module.params['num_ports']
+        pg_spec = vim.dvs.DistributedVirtualPortgroup.ConfigSpec()
 
-        # Default port config
-        config.defaultPortConfig = vim.dvs.VmwareDistributedVirtualSwitch.VmwarePortConfigPolicy()
-        if self.module.params['vlan_trunk']:
-            config.defaultPortConfig.vlan = vim.dvs.VmwareDistributedVirtualSwitch.TrunkVlanSpec()
-            vlan_id_list = []
-            for vlan_id_splitted in self.module.params['vlan_id'].split(','):
-                try:
-                    vlan_id_start, vlan_id_end = vlan_id_splitted.split('-')
-                    vlan_id_list.append(vim.NumericRange(start=int(vlan_id_start.strip()), end=int(vlan_id_end.strip())))
-                except ValueError:
-                    vlan_id_list.append(vim.NumericRange(start=int(vlan_id_splitted.strip()), end=int(vlan_id_splitted.strip())))
-            config.defaultPortConfig.vlan.vlanId = vlan_id_list
+        # Name
+        results['portgroup'] = self.portgroup_name
+        pg_spec.name = self.portgroup_name
+        # Description
+        results['description'] = self.description
+        pg_spec.description = self.description
+        # Number of ports
+        results['num_ports'] = self.num_ports
+        pg_spec.numPorts = self.num_ports
+        # Port binding
+        # NOTE: 'self.portgroup_type' is deprecated. Will be removed in 2.11
+        if self.portgroup_type:
```

Comment:
So if both `portgroup_type` and `port_binding` are defined, it ignores `port_binding`? I think it should check for this and if they are not the same from the API perspective, it should fail with an error

Reply:
Does the UI offer both options? And it ignores one if the other is set?

Reply:
Ok. Now I know what you mean. `portgroup_type` is for backward compatibility. I don't want that the module fails if it's specified. Otherwise, I could just remove it since the user needs to change his play anyway..

Reply:
Yeah, i understand the backward compatibility, can you add in the description of `port_binding` that it takes precedence over the deprecated `portgroup_type` value? That way people understand the precedence.